lib/src/_network_image_io.dart

```diff
@@ -149,29 +150,22 @@ class ExtendedNetworkImageProvider
   Future<Uint8List> _loadCache(
     ExtendedNetworkImageProvider key,
     StreamController<ImageChunkEvent> chunkEvents,
-    String md5Key,
+    String url,
   ) async {
-    final Directory _cacheImagesDirectory = Directory(
-        join((await getTemporaryDirectory()).path, cacheImageFolderName));
-    //exist, try to find cache image file
-    if (_cacheImagesDirectory.existsSync()) {
-      final File cacheFlie = File(join(_cacheImagesDirectory.path, md5Key));
-      if (cacheFlie.existsSync()) {
-        return await cacheFlie.readAsBytes();
-      }
-    }
-    //create folder
-    else {
-      await _cacheImagesDirectory.create();
+    final GetOrSetCacheImageResult result =
+        await network_utils.getOrSetCachedImageFile(url);
+    if (result.data != null) {
+      return result.data;
     }
+
     //load from network
     final Uint8List data = await _loadNetwork(
       key,
       chunkEvents,
     );
     if (data != null) {
       //cache image file
-      await File(join(_cacheImagesDirectory.path, md5Key)).writeAsBytes(data);
```

Comment:
waiting cache file writing reduces performance, I removed await, but if suddenly I'm not aware of something, then tell me